### ISSUE TITLE
byconity-server become non-responsive after a while

### ISSUE THREAD

**Issue by lazywei** [ByConity/ByConity#909]
## Bug Report

`byconity-server` became non-responsive after a while. i.e., the liveness probe and the readiness probe fails on k8s.
I tried manually connecting to the server and it seems it just doesn't response
```
clickhouse --user default --password FOOOBAR
```
will just hang there indefinitely.

In the meantime, the server doesn't seem to product any meaningful log, here is the logs we got during this non-responsive time.
Is there anything specific or suspicious about these logs? Like is it normal to see the "Yield" status on the previous leader over and over again?

Not sure if this is due to any HA or leader selection issue because the server is not consuming a lot of CPU / RAM. It seems it enter some bad locking state and stop accepting any request.

```
2023.12.01 00:15:06.218575 [ 2487 ] {} <Trace> TCPHandlerFactory: TCP Request. Address: [::ffff:10.64.94.104]:64877
2023.12.01 00:15:06.218666 [ 2487 ] {} <Information> TCPHandler: Client has not sent any data.
2023.12.01 00:15:06.218678 [ 2487 ] {} <Debug> TCPHandler: Done processing connection.
2023.12.01 00:15:06.561440 [ 2487 ] {} <Trace> TCPHandlerFactory: TCP Request. Address: [::1]:49068
2023.12.01 00:15:06.561527 [ 2487 ] {} <Debug> TCPHandler: Connected ByConity client version 21.8.0, revision: 54450, user: default.
2023.12.01 00:15:06.561538 [ 2487 ] {} <Debug> TCPHandler: Connected default
2023.12.01 00:15:06.621379 [ 2522 ] {} <Trace> TCPHandlerFactory: TCP Request. Address: [::ffff:10.64.248.224]:63368
2023.12.01 00:15:06.621532 [ 2522 ] {} <Information> TCPHandler: Client has not sent any data.
2023.12.01 00:15:06.621546 [ 2522 ] {} <Debug> TCPHandler: Done processing connection.
2023.12.01 00:15:06.944770 [ 242 ] {} <Trace> StorageElector: It is refreshed successfully and not expired: address {
  hostname: "10.64.217.123"
  host: "10.64.217.123"
  rpc_port: 8124
  tcp_port: 9000
  http_port: 8123
  exchange_port: 8124
  exchange_status_port: 8124
}
lease {
  elected_time: 1701387339213
  last_refresh_time: 1701389706941
  refresh_interval_ms: 1000
  expired_interval_ms: 5000
  status: Ready
}
previous_leader_address {
  hostname: "10.64.199.112"
  host: "10.64.199.112"
  rpc_port: 8124
  tcp_port: 9000
  http_port: 8123
  exchange_port: 8124
  exchange_status_port: 8124
}
previous_leader_lease {
  elected_time: 1701387142595
  last_refresh_time: 1701387144602
  refresh_interval_ms: 1000
  expired_interval_ms: 5000
  status: Yield
}

2023.12.01 00:15:07.948120 [ 242 ] {} <Trace> StorageElector: It is refreshed successfully and not expired: address {
  hostname: "10.64.217.123"
  host: "10.64.217.123"
  rpc_port: 8124
  tcp_port: 9000
  http_port: 8123
  exchange_port: 8124
  exchange_status_port: 8124
}
lease {
  elected_time: 1701387339213
  last_refresh_time: 1701389707944
  refresh_interval_ms: 1000
  expired_interval_ms: 5000
  status: Ready
}
previous_leader_address {
  hostname: "10.64.199.112"
  host: "10.64.199.112"
  rpc_port: 8124
  tcp_port: 9000
  http_port: 8123
  exchange_port: 8124
  exchange_status_port: 8124
}
previous_leader_lease {
  elected_time: 1701387142595
  last_refresh_time: 1701387144602
  refresh_interval_ms: 1000
  expired_interval_ms: 5000
  status: Yield
}
```

**Comment by lazywei** [comment]
The topology we got from FDB:
```
fdb> get byconity_SERVERS_TOPOLOGY
`byconity_SERVERS_TOPOLOGY' is `\x0a\x82\x01\x08\xde\xc7\x8c\x96\xc21\x1ap\x0a\x11server_vw_default\x12[\x0aA10-64-217-123.byconity-server-headless.byconity.svc.cluster.local\x12\x0d10.64.217.123\x18\xbc? \xa8F(\xbb? \xe1\xe3\x88\x96\xc21(\x03'
```

**Comment by lazywei** [comment]
How can we dump the stack? is there a way to trigger that? because the server just act normal (continue to log info/debug/trace etc.) without crashing but the HTTP and TCP endpoint is non-responsive.

**Comment by lazywei** [comment]
We added the `use_minidump: true` to the `cnch_config.yaml`
![image](https://github.com/ByConity/ByConity/assets/2840571/1b4b1353-b5cb-4b6b-ae8d-d626e7b6b144)

But after sending the signal to the server, we only see this line and there is no other information being given and didn't see any dump file being generated either
```
root@byconity-server-0:/opt/byconity# tail -f /var/log/byconity/byconity.log | grep minidump
2023.12.01 06:32:29.322999 [ 50 ] {} <Debug> BaseDaemon: Received signal to minidump
```

Is there other way to verify the minidump is actually dumping something?

**Comment by GallifreyY** [comment]
is there any dmp file under log path? @lazywei

**Comment by lazywei** [comment]
This is all we got under the log path
```
root@byconity-server-0:/var/log/byconity# ls
byconity.err.log  byconity.log	byconity.log.0.gz  byconity.log.1.gz  lost+found
```

**Comment by GallifreyY** [comment]
> This is all we got under the log path
> 
> ```
> root@byconity-server-0:/var/log/byconity# ls
> byconity.err.log  byconity.log	byconity.log.0.gz  byconity.log.1.gz  lost+found
> ```

we didn't enable breakpad that's why minidump can not generated. we will enable it later and update u asap. cc @dmthuc

**Comment by lazywei** [comment]
gotcha i will do that and see if i can find anything, thanks!

**Comment by lazywei** [comment]
I'm now able to get the dump file but is `minidump-2-core` binary also included in the image? cuz I couldn't find this binary to convert the dump to core.

**Comment by lazywei** [comment]
is there an easy way for me to include the mindump-2-core into the build target? I'm already building from the source with debian docker so if it's an easy one-liner I can probably do it my on own :)

**Comment by lazywei** [comment]
I got these warnings but I was able to get the `gdb.thread` file
link to the file https://gist.github.com/lazywei/10aabc66b9354c2a48263a7760cc7319


```
warning: core file may not match specified executable file.
Warning: couldn't activate thread debugging using libthread_db: Cannot find new threads: generic error

warning: File "/lib/x86_64-linux-gnu/libthread_db-1.0.so" auto-loading has been declined by your `auto-load safe-path' set to "$debugdir:$datadir/auto-load".

warning: Unable to find libthread_db matching inferior's thread library, thread debugging will not be available.
Failed to read a valid object file image from memory.
Warning: couldn't activate thread debugging using libthread_db: Cannot find new threads: generic error

warning: File "/lib/x86_64-linux-gnu/libthread_db-1.0.so" auto-loading has been declined by your `auto-load safe-path' set to "$debugdir:$datadir/auto-load".

warning: Unable to find libthread_db matching inferior's thread library, thread debugging will not be available.
```

**Comment by vivek-debug** [comment]
@lazywei @GallifreyY This is an old bug from RBAC that will be fixed in #916
